js/classes/entities/foreground/dialog_bubble.js

```diff
@@ -34,6 +42,164 @@ class DialogBubble {
         };
     };
 
+    /**
+     * Update is going to add a single character of text to the bubble until all of them have been added, then nothing.
+     */
+    update() {
+        // It only ever updates when you haven't added every line.
+        if (this.currentLine < this.targetLines.length) {
+            // You're gonna keep adding to the same line.
+            if (this.visibleLines[this.currentLine].length < this.targetLines[this.currentLine].length) {
+                const newChar = this.targetLines[this.currentLine].charAt(this.visibleLines[this.currentLine].length);
+                this.visibleLines[this.currentLine] = this.visibleLines[this.currentLine] + newChar;
+            } else {
+                this.currentLine++;
+            }
+        }
+    };
+
+    /**
+     * This method is going to draw the dialog bubble, the speaker, and the text (in this.visibleLines).
+     */
+    draw() {
+        // three steps:
+        const dbPos = new Vector(
+            (Camera.SIZE.x - DialogBubble.SCALED_SIZE.x) / 2,
+            Camera.SIZE.y - DialogBubble.SCALED_SIZE.y
+        );
+
+        // (1) Draw the DialogBubble sprite.
+        CTX.drawImage(
+            ASSET_MGR.getAsset(DialogBubble.SPRITESHEET),
+            0, DialogBubble.SIZE.y * this.type,
+            DialogBubble.SIZE.x, DialogBubble.SIZE.y,
+            dbPos.x, dbPos.y,
+            DialogBubble.SCALED_SIZE.x, DialogBubble.SCALED_SIZE.y);
+            
+        // (2) Draw the speaker.
+        const scale = 6;
+        const speakerStart = this.speaker.spritesheet === DialogBubble.SPEAKERS.CHAD.spritesheet ?
+            new Vector(
+                ((Camera.SIZE.x - DialogBubble.SCALED_SIZE.x) / 2) - (this.speaker.size.x * 0.5),
+                Camera.SIZE.y - DialogBubble.SCALED_SIZE.y
+            ) : new Vector(
+                Camera.SIZE.x - ((Camera.SIZE.x - DialogBubble.SCALED_SIZE.x) / 2) + (this.speaker.size.x * 0.5),
+                Camera.SIZE.y - DialogBubble.SCALED_SIZE.y
+            );
+        this.speaker.drawFrame(speakerStart, scale);
+
+        // (3) Draw the text.
+        let textStart = new Vector(
+            (Camera.SIZE.x - DialogBubble.SCALED_SIZE.x) / 2 + (0.1 * DialogBubble.SCALED_SIZE.x),
+            Camera.SIZE.y - DialogBubble.SCALED_SIZE.y + (0.1 * DialogBubble.SCALED_SIZE.x) - (5 * DialogBubble.SCALE) + 17
+        );
+        // The jump in pixels after each line!
+        const textJump = new Vector(0, 40); // NOTE: this is based off a text size of 34!
```

Comment:
Maybe make this a static constant as well?

Reply:
I could, but it's only relevant in a single method so wouldn't really be necessary I think.